packages/x-components/src/components/location-provider.vue

```diff
@@ -0,0 +1,59 @@
+<script lang="ts">
+  import { Prop, Provide, Component } from 'vue-property-decorator';
+  import { NoElement } from '../components/no-element';
+  import { FeatureLocation } from '../types';
+
+  /**
+   * Location Provider component.
+   * This component injects the location with value passed as prop.
+   *
+   * @public
+   */
+  @Component
+  export default class LocationProvider extends NoElement {
+    /**
+     * The {@link FeatureLocation} to provide.
+     *
+     * @public
+     */
+    @Prop({ required: true })
+    @Provide()
+    protected location!: FeatureLocation;
+  }
+</script>
+
+<docs lang="mdx">
+## Play with props
+
+In this example, the location provider receives a prop with the location to provide to its child
+components.
+
+```vue
+<template>
+  <LocationProvider location="external">
+    <BaseEventButton :events="events" />
+  </LocationProvider>
+</template>
+
+<script>
+  import { BaseEventButton, LocationProvider } from '@empathyco/x-components';
+
+  export default {
+    name: 'LocationProviderDemo',
+    components: {
+      BaseEventButton,
+      LocationProvider
+    },
+    data() {
+      return {
+        events: ['UserAcceptedAQuery']
+      };
+    }
+  };
+</script>
+```
+
+## Events
+
+This component doesn't emit events.
+</docs>
```

Comment:
A few advices that will improve the doc IMO: 

1. Lets try to follow the same structure for new components than the ones that have been validated by eDocs, like the one in `SearchInput`.
2. `BaseEventButton` component does not expect an array of events, but an object. The `UserAcceptedAQuery` event should always have a payload. In any case, I would use a real example.
3. This examples are supposedly going to be rendered, as this component is not very visual, and it all about the metadata in the events, let's try to leave it crystal clear why you need this.

This is my proposal:

```suggestion
## Events

This component doesn't emit events. However, events emitted by child components, will include the
`location` prop value in its metadata.

## See it in action

In this example, the `LocationProvider` component receives the `location` prop with the
`"predictive_layer"` value. This value will then be combined with the `NextQueries` feature name,
`next_query` to create a `QueryOrigin` string, which is used for tracking the performance of the
different tools that generate the query.

```vue
<template>
  <LocationProvider location="predictive_layer">
    <NextQueries :events="events" />
  </LocationProvider>
</template>

<script>
  import { LocationProvider } from '@empathyco/x-components';
  import { NextQueries } from '@empathyco/x-components/next-queries';

  export default {
    name: 'LocationProviderDemo',
    components: {
      NextQueries,
      LocationProvider
    }
  };
</script>
```
```

Reply:
You're completely right, I'll remove the events from the NextQueries element

`<NextQueries :events="events" />`

as the rest s perfect.